src/modules/reporting/containers/reporting-dispute.js

```diff
@@ -2,17 +2,23 @@ import { connect } from 'react-redux'
 import { withRouter } from 'react-router-dom'
 
 import ReportingDispute from 'modules/reporting/components/reporting-dispute/reporting-dispute'
+import makePath from 'src/modules/routes/helpers/make-path'
+import { ACCOUNT_DEPOSIT } from 'src/modules/routes/constants/views'
+import { selectLoginAccount } from 'src/modules/auth/selectors/login-account'
 
-const mapStateToProps = state => ({
-  markets: [],
-  marketsCount: 0,
-  isMobile: state.isMobile,
-})
+const mapStateToProps = (state, { history }) => {
+  const loginAccount = selectLoginAccount(state)
+  return ({
+    doesUserHasRep: (loginAccount.rep.value > 0),
+    markets: [],
+    marketsCount: 0,
+    isMobile: state.isMobile,
+    navigateToAccountDepositHandler: () => history.push(makePath(ACCOUNT_DEPOSIT)),
```

Comment:
Instead of passing this, refactor the component's `button` to be a `react-router-dom` `Link` component.